uaclient/schema/status_1.py

```diff
@@ -0,0 +1,132 @@
+"""
+Schema v.1: consolidate account and contract keys.
+
+This schema version is incompatible with previous status.json schema because:
+ - account key is now a dict instead of string to collect related id, external_account_ids and name under a single object
+ - subscription-id and subscription keys are dropped
+
+General notes:
+
+   We will increment a schema version for incompatible changes:
+    - changing the value type for existing keys
+    - dropping deprecated keys
+
+   Each schema patch file will:
+    - be named <schema_version_number>.py  # schema_version_number is an int
+    - contain up() and down() functions to adapt existing status values to the
+      desired incremental or decremental schema version. 
+    - the functions up and down will return the updated schema dict
+"""
+import copy
+import re
+import yaml
+
+from uaclient.config import UAConfig
+
+from typing import Any, Dict
+
+NEW_KEYS = ("contract",)
+DROPPED_KEYS = ("subscription", "subscription-id")
+CHANGED_KEYS = ("account",)  # keys where the value type has changed
+
+PATCH_VERSION = float(re.sub(r".*status_([.\d]+).py", r"\1", __file__))
+
+
+def up(cfg: UAConfig, current_status: Dict) -> Dict:
+    """Patch current_status upgrading from previous schema version status dict.
+
+    :return: Dict of updated config
+    """
+    new_status = copy.deepcopy(current_status)
+    # Asserts to make sure we are looking at the expected current_status
+    for key in NEW_KEYS:
+        assert (
+            key not in current_status
+        ), "Found unexpected key: {} cannot patch status.json.".format(key)
+    for key in DROPPED_KEYS + CHANGED_KEYS:
+        assert (
+            key in current_status
+        ), "Missing expected key: {}. Cannot patch status.json".format(key)
+        new_status.pop(key)
+    new_status["_schema_version"] = PATCH_VERSION
+    if current_status["attached"]:
+        token = cfg.machine_token["machineTokenInfo"]
+        account_info = token["accountInfo"]
```

Comment:
we could use `cfg.accounts` here, right ?

Reply:
Yes we could. but we need to be explicit about what specific keys we are using during the patch up or down because tip of `UAConfig.accounts` could drift as we patch the schema to newer versions, so it may not be able to reference the key properly once tip moves to schema version 2 or 3 because the upstream key may have changed. We can only rely on the key as defined in the current version we are patching.